modular_convert

hsz06 opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/PaddleFormers/pull/ -->
#### Before submitting

- [ ] Lint code. If there are lint issues, please format the code first.

```shell
# Install and register `pre-commit` in the project folder
pip install pre-commit && pre-commit install

# Process previous code files separately
pre-commit run --file XXXX.py
```

- [ ] Add test cases into `tests` folder. If there are codecov issues, please add tests cases first.

### PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->

### PR changes
<!-- One of [ Models | APIs | Docs | Others ] -->

### Description
<!-- Describe what this PR does -->

hardtochooseaname committed:
new file:   paddleformers/convert/doc/README.md
	new file:   paddleformers/convert/doc/f9950dee47aab4918e0dc9702f21cb01.png
	new file:   paddleformers/convert/example/convert.py
	new file:   paddleformers/convert/example/fix_import.py
	new file:   paddleformers/convert/example/modular_qwen2.py

hardtochooseaname committed:
new file:   "paddleformers/convert/doc/llama\346\236\266\346\236\204\345\210\206\346\236\220.md"
	new file:   "paddleformers/convert/doc/qwen2\346\236\266\346\236\204\345\210\206\346\236\220.md"

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PaddlePaddle/PaddleFormers?pullRequest=2430) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/PaddlePaddle/PaddleFormers?pullRequest=2430) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: hsz06<br/>:x: hardtochooseaname<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PaddlePaddle/PaddleFormers?pullRequest=2430) it.</sub>

hardtochooseaname committed:
renamed:    paddleformers/convert/doc/README.md -> paddleformers/transformers/convert/doc/README.md
	renamed:    paddleformers/convert/doc/f9950dee47aab4918e0dc9702f21cb01.png -> paddleformers/transformers/convert/doc/f9950dee47aab4918e0dc9702f21cb01.png
	renamed:    "paddleformers/convert/doc/llama\346\236\266\346\236\204\345\210\206\346\236\220.md" -> "paddleformers/transformers/convert/doc/llama\346\236\266\346\236\204\345\210\206\346\236\220.md"
	renamed:    "paddleformers/convert/doc/qwen2\346\236\266\346\236\204\345\210\206\346\236\220.md" -> "paddleformers/transformers/convert/doc/qwen2\346\236\266\346\236\204\345\210\206\346\236\220.md"
	renamed:    paddleformers/convert/example/convert.py -> paddleformers/transformers/convert/example/convert.py
	renamed:    paddleformers/convert/example/fix_import.py -> paddleformers/transformers/convert/example/fix_import.py
	renamed:    paddleformers/convert/example/modular_qwen2.py -> paddleformers/transformers/convert/example/modular_qwen2.py

hsz06 committed:
new file:   paddleformers/transformers/convert/configuration.py
	modified:   paddleformers/transformers/convert/doc/README.md
	deleted:    paddleformers/transformers/convert/example/convert.py
	deleted:    paddleformers/transformers/convert/example/fix_import.py
	new file:   paddleformers/transformers/convert/example/hugeface/modeling.py
	new file:   paddleformers/transformers/convert/example/hugeface/modeling_llama.py
	new file:   paddleformers/transformers/convert/example/hugeface/modular_qwen.py
	deleted:    paddleformers/transformers/convert/example/modular_qwen2.py
	new file:   paddleformers/transformers/convert/main.py
	new file:   paddleformers/transformers/convert/modeling_imports_results.txt
	new file:   paddleformers/transformers/convert/test_ability/test_base/test_modeling.py
	new file:   paddleformers/transformers/convert/test_model_expanded.py
	new file:   paddleformers/transformers/convert/test_modular.py
	new file:   paddleformers/transformers/convert/until/collect_import_modeling.py
	new file:   paddleformers/transformers/convert/until/rename_identifiers.py
	new file:   paddleformers/transformers/convert/until/rewrite_child_classes.py

DrownFish19 commented on paddleformers/transformers/convert/example/hugeface/modeling.py:
这个文件夹的文件可以删除

DrownFish19 commented on paddleformers/transformers/convert/modeling_imports_results.txt:
中间文件可以删除

DrownFish19 commented on paddleformers/transformers/qwen2/modeling_qwen2.py:
代码正式化到qwen2目录下

DrownFish19 replied:
这里需要指定模型文件名为modeling.py，兼容paddleformers内容。

DrownFish19 commented on paddleformers/transformers/convert/test_modular.py:
代码正式化到qwen2目录下

DrownFish19 commented on paddleformers/transformers/convert/configuration.py:
此文件可以删除

DrownFish19 commented on paddleformers/transformers/convert/doc/README.md:
补充以下关键章节
1. 使用方式
2. 和huggingface使用方式的差异点
3. 以qwen2为例的模型转换精度对齐说明

DrownFish19 commented:
1. 补充 modular 单测，确保modular转换方式正确。
2. 转换方式应增加至pre-commit，每次提交代码前自动进行转换验证。参考其他功能prec-commit实现：https://github.com/PaddlePaddle/PaddleFormers/blob/dc9ac0e62997a0b671439fd7a018777bf418a79e/.pre-commit-config.yaml#L47-L54

hsz06 committed:
deleted:    paddleformers/transformers/convert/configuration.py
	new file:   "paddleformers/transformers/convert/doc/hugeface\350\275\254\346\215\242\345\210\206\346\236\220.md"
	new file:   "paddleformers/transformers/convert/doc/\347\262\276\345\272\246\345\257\271\351\275\220_\345\271\266\350\241\214\350\203\275\345\212\233.md"
	deleted:    paddleformers/transformers/convert/example/hugeface/modeling.py
	deleted:    paddleformers/transformers/convert/example/hugeface/modeling_llama.py
	deleted:    paddleformers/transformers/convert/example/hugeface/modular_qwen.py
	modified:   paddleformers/transformers/convert/main.py
	deleted:    paddleformers/transformers/convert/modeling_imports_results.txt
	new file:   paddleformers/transformers/convert/test_ability/test_base/test.py
	modified:   paddleformers/transformers/convert/test_ability/test_base/test_modeling.py
	new file:   paddleformers/transformers/convert/test_ability/test_paralle/compare_torch_with_paddle.py
	deleted:    paddleformers/transformers/convert/test_modular.py
	modified:   paddleformers/transformers/convert/until/collect_import_modeling.py
	modified:   paddleformers/transformers/convert/until/rename_identifiers.py
	modified:   paddleformers/transformers/convert/until/rewrite_child_classes.py
	modified:   paddleformers/transformers/qwen2/__init__.py
	modified:   paddleformers/transformers/qwen2/configuration.py
	new file:   paddleformers/transformers/qwen2/fusion_ops.py
	renamed:    paddleformers/transformers/convert/test_model_expanded.py -> paddleformers/transformers/qwen2/modeling_qwen2.py
	new file:   paddleformers/transformers/qwen2/modular_qwen2.py

hsz06 committed:
new file:   "paddleformers/transformers/convert/doc/\344\275\277\347\224\250\346\226\271\346\263\225.md"

DrownFish19 commented on paddleformers/transformers/convert/doc/hugeface转换分析.md:
文件名修改，不使用中文
“huggingface_modular_convert.md”

DrownFish19 commented on paddleformers/transformers/convert/doc/使用方法.md:
README.md

DrownFish19 commented on paddleformers/transformers/convert/doc/精度对齐_并行能力.md:
precision.md

hsz06 committed:
deleted:    paddleformers/transformers/convert/doc/README.md

hsz06 committed:
renamed:    "paddleformers/transformers/convert/doc/\344\275\277\347\224\250\346\226\271\346\263\225.md" -> paddleformers/transformers/convert/doc/README.md
	deleted:    paddleformers/transformers/convert/doc/f9950dee47aab4918e0dc9702f21cb01.png
	deleted:    "paddleformers/transformers/convert/doc/hugeface\350\275\254\346\215\242\345\210\206\346\236\220.md"
	renamed:    "paddleformers/transformers/convert/doc/llama\346\236\266\346\236\204\345\210\206\346\236\220.md" -> paddleformers/transformers/convert/doc/llama_analysis.md
	new file:   paddleformers/transformers/convert/doc/process.png
	renamed:    "paddleformers/transformers/convert/doc/qwen2\346\236\266\346\236\204\345\210\206\346\236\220.md" -> paddleformers/transformers/convert/doc/qwen2_analysis.md
	deleted:    "paddleformers/transformers/convert/doc/\347\262\276\345\272\246\345\257\271\351\275\220_\345\271\266\350\241\214\350\203\275\345\212\233.md"
	new file:   "paddleformers/transformers/convert/doc/\351\241\271\347\233\256\346\212\245\345\221\212\357\274\232\351\243\236\346\241\250PaddleNLP-\345\211\215\346\262\277\346\250\241\345\236\213\346\250\241\345\235\227\345\214\226\350\256\276\350\256\241.md"
	deleted:    paddleformers/transformers/convert/test_ability/test_base/test.py
	modified:   paddleformers/transformers/qwen2/modeling.py

huanghengheng commented:
CI长时间pending状态，建议close

hsz06 committed:
modified:   paddleformers/transformers/convert/doc/README.md
